arwen/contexts/managedType.go

```diff
@@ -313,103 +306,103 @@ func (context *managedTypesContext) GetPrivateKeyByteLengthEC(ecHandle int32) in
 
 // NewManagedBuffer creates a new empty buffer in the managed buffers map and returns the handle
 func (context *managedTypesContext) NewManagedBuffer() int32 {
-	newHandle := int32(len(context.manBufValues))
+	newHandle := int32(len(context.managedTypesValues.mBufferValues))
 	for {
-		if _, ok := context.manBufValues[newHandle]; !ok {
+		if _, ok := context.managedTypesValues.mBufferValues[newHandle]; !ok {
 			break
 		}
 		newHandle++
 	}
-	newManBuf := make([]byte, 0)
-	context.manBufValues[newHandle] = newManBuf
+	newmBuffer := make([]byte, 0)
+	context.managedTypesValues.mBufferValues[newHandle] = newmBuffer
 	return newHandle
 }
 
 // NewManagedBufferFromBytes creates a new buffer in the managed buffers map, sets the bytes provided, and returns the handle
 func (context *managedTypesContext) NewManagedBufferFromBytes(bytes []byte) int32 {
-	manBufHandle := context.NewManagedBuffer()
-	context.SetBytesForThisManagedBuffer(manBufHandle, bytes)
-	return manBufHandle
+	mBufferHandle := context.NewManagedBuffer()
+	context.SetBytesForThisManagedBuffer(mBufferHandle, bytes)
+	return mBufferHandle
 }
 
 // SetBytesForThisManagedBuffer sets the bytes given as value for the managed buffer. Returns 0 if success, 1 otherwise
-func (context *managedTypesContext) SetBytesForThisManagedBuffer(manBufHandle int32, bytes []byte) int32 {
-	_, ok := context.manBufValues[manBufHandle]
+func (context *managedTypesContext) SetBytesForThisManagedBuffer(mBufferHandle int32, bytes []byte) bool {
+	_, ok := context.managedTypesValues.mBufferValues[mBufferHandle]
```

Comment:
in case of Set functions I think we could work with GetOrCreate - thus creating the managed buffer if it did not exist.
The same with GetArgument to managed buffer

Reply:
I am more in favor of only working with explicit buffer creation, since it is safer.
Trying to get or set to a non-existing one should be an error.